### PR TITLE
COMMONS-547 A Null pointer exception while processing UINotificationPopoverToolba…

### PR DESCRIPTION
**Pull request by AtiaZ**
…rPortlet in cluster prevents second node to start

### CODE REVIEW

File: extension/portlets/platformNavigation/src/main/java/org/exoplatform/platform/component/UINotificationPopoverToolbarPortlet.java [hunk 1]
```diff
@@ -64,58 +65,85 @@ public UINotificationPopoverToolbarPortlet() throws Exception {
...
-           .append(getCometdContextName()).append("');");
-    
-    context.getJavascriptManager().getRequireJS()
-           .require("SHARED/jquery_cometd", "cometd")
-           .require("PORTLET/platformNavigation/NotificationPopoverToolbarPortlet", "NotificationPopoverToolbarPortlet")
-           .addScripts(scripts.toString());
-    //
-    super.processRender(app, context);
+    if(this.currentUser != null) {
+      String currentUserId = getCurrentUserId();
```
Comment (aboughzela) [inline extension/portlets/platformNavigation/src/main/java/org/exoplatform/platform/component/UINotificationPopoverToolbarPortlet.java]: When context.getRemoteUser() not null is not necessary to get  currentUser from ConversationState

File: extension/portlets/platformNavigation/src/main/java/org/exoplatform/platform/component/UINotificationPopoverToolbarPortlet.java [hunk 2]
```diff
@@ -64,58 +65,85 @@ public UINotificationPopoverToolbarPortlet() throws Exception {
...
-      int badge = clusteringProcess(notificationId);
-      if(badge > 0) {
-        MimeResponse res = context.getResponse();
-        res.setContentType("application/json");
-        JSONObject object = new JSONObject();
-        object.put("badge", String.valueOf(badge));
-        res.getWriter().write(object.toString());
+    this.currentUser = context.getRemoteUser();
+    if (this.currentUser != null) {
+      String currentUserId = getCurrentUserId();
```
Comment (aboughzela) [inline extension/portlets/platformNavigation/src/main/java/org/exoplatform/platform/component/UINotificationPopoverToolbarPortlet.java]: When context.getRemoteUser() not null is not necessary to get currentUser from ConversationState

File: extension/portlets/platformNavigation/src/main/java/org/exoplatform/platform/component/UINotificationPopoverToolbarPortlet.java (hunk 1 above)
Comment (AtiaZ) [reply extension/portlets/platformNavigation/src/main/java/org/exoplatform/platform/component/UINotificationPopoverToolbarPortlet.java]: change added as discussed

File: extension/portlets/platformNavigation/src/main/java/org/exoplatform/platform/component/UINotificationPopoverToolbarPortlet.java (hunk 2 above)
Comment (AtiaZ) [reply extension/portlets/platformNavigation/src/main/java/org/exoplatform/platform/component/UINotificationPopoverToolbarPortlet.java]: change added as discussed

**Pull request merged**